File: WorkcraftCore/src/main/java/org/workcraft/dom/visual/connections/ConnectionGui.java
package org.workcraft.dom.visual.connections;

import org.workcraft.dom.visual.ColorisableGraphicalContent;
import org.workcraft.dom.visual.Touchable;
import org.workcraft.util.Function;

public interface ConnectionGui {
	
	public static Function<ConnectionGui, ColorisableGraphicalContent> getGraphicalContent = new Function<ConnectionGui, ColorisableGraphicalContent>(){
		@Override
		public ColorisableGraphicalContent apply(ConnectionGui argument) {
			return argument.graphicalContent();
		}
	};
	
	public static Function<ConnectionGui, Touchable> getTouchable = new Function<ConnectionGui, Touchable>(){
		@Override
		public Touchable apply(ConnectionGui argument) {
			return argument.shape();
		}
	};
	
	Touchable shape();
	ColorisableGraphicalContent graphicalContent();
	ParametricCurve parametricCurve();
}

File: ScalaGraphEditorUtil/src/main/scala/org/workcraft/dom/visual/connections/ConnectionGui.scala
package org.workcraft.dom.visual.connections
import org.workcraft.graphics.Touchable
import org.workcraft.graphics.ColorisableGraphicalContent
import org.workcraft.graphics.TouchableC

case class ConnectionGui(
    shape : TouchableC
    , graphicalContent : ColorisableGraphicalContent
    , parametricCurve : org.workcraft.graphics.ParametricCurve)
